implementations/rust/ockam/ockam_api/src/util.rs

```diff
@@ -54,56 +54,75 @@ pub async fn multiaddr_to_route(
 ) -> Option<MultiAddrToRouteResult> {
     let mut rb = Route::new();
     let mut it = ma.iter().peekable();
-    let flow_control_id = flow_controls.generate_id();
 
-    let mut options = Some(TcpConnectionOptions::as_producer(
-        flow_controls,
-        &flow_control_id,
-    ));
+    let mut flow_control_id = None;
+    let mut number_of_tcp_hops = 0;
 
     while let Some(p) = it.next() {
         match p.code() {
             Ip4::CODE => {
+                if number_of_tcp_hops >= 1 {
```

Comment:
This condition could be lifted out of the `match`

Reply:
Not sure

Reply:
We want the condition only for dns,ipv4 and ipv6

Reply:
Sorry I skipped the other cases with code folding. I wish we had some common abstraction for those 3 cases but maybe there's nothing that really fits.